### PR TITLE
[fix](Nerieds) fix FoldConstantRuleOnFE too slow when CaseWhen clause nesting multiple layers

### PR DESCRIPTION
**Pull request by 924060929**
## Proposed changes

fix FoldConstantRuleOnFE too slow when CaseWhen clause nesting multiple layers.

### CODE REVIEW

**Comment by 924060929** [comment]
run buildall

**Review by morrySnow — DISMISSED** [review]

**Comment by 924060929** [comment]
run buildall

File: regression-test/suites/nereids_syntax_p0/slow_fold_constant_case_when.groovy [hunk 1]
```diff
@@ -0,0 +1,824 @@
...
+                    AND _____ != '______' ) t
+            GROUP BY  grouping sets((_____) ,())
+            ORDER BY  _____ nulls last limit 3000 ) t
+        ORDER BY  _____ nulls last
+            """
+
+    sql "set enable_fold_constant_by_be=false"
+    test {
+        sql sqlStr
+        time 60000
```
Comment (zclllyybb) [inline regression-test/suites/nereids_syntax_p0/slow_fold_constant_case_when.groovy:816]: is it acceptable in P0 rather than P1?

Comment (924060929) [reply regression-test/suites/nereids_syntax_p0/slow_fold_constant_case_when.groovy:816]: It should be fast, so we don't need to move to p1

**Review by morrySnow — APPROVED** [review]

**Review by starocean999 — APPROVED** [review]

**Pull request merged**